http/src/main/java/com/hazelcast/jet/contrib/http/impl/HttpListenerSinkContext.java

```diff
@@ -117,8 +118,11 @@ private void drainBuffer() {
         if (messageBuffer == null) {
             return;
         }
-        messageBuffer.forEach(sinkHttpHandler::send);
-        messageBuffer.clear();
+        String message = messageBuffer.poll();
+        while (message != null) {
+            sinkHttpHandler.send(message);
+            message = messageBuffer.poll();
+        }
     }
 
     private void putToBuffer(T item) {
```

Comment:
Do I understand this correctly? For example, if we buffer 10 items and there's no client connected, we'll accumulate the first 10 items and throw away the rest. When a client connects, we'll send the old 10 items? Sounds weird.

Reply:
I've changed the logic so that last 10 items are returned to the client